products/swift-bindings/CMakeLists.txt

```diff
@@ -1,2 +1,83 @@
-add_custom_target(swift-bindings
-                  DEPENDS libllbuild)
+# FIXME: This function should probably be in cmake/modules/Utility.cmake
+# FIXME: Incremental builds.  
+# Compile swift sources to swiftmodules.
+# Usage:
+# target          # Target name
+# name            # Swift Module name
+# deps            # Target dependencies
+# sources         # List of sources
+# additional_args # List of additional args to pass
+function(add_swift_module target name deps sources additional_args)
+
+  list(APPEND ARGS -module-name)
+  list(APPEND ARGS ${name})
+  list(APPEND ARGS -incremental -emit-dependencies)
+  list(APPEND ARGS -emit-module)
+  list(APPEND ARGS -emit-module-path)
+  list(APPEND ARGS ${name}.swiftmodule)
+  
+  set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+  set(FILEMAP ${BUILD_DIR}/output-file-map.json)
+  
+  file(REMOVE ${FILEMAP})
+  file(APPEND ${FILEMAP} "{\n\"\": {\n")
+  file(APPEND ${FILEMAP} "\"swift-dependencies\": \"${BUILD_DIR}/master.swiftdeps\"\n")
+  file(APPEND ${FILEMAP} "},\n")
+  
+  foreach(source ${sources})
+    file(APPEND ${FILEMAP} "\"${CMAKE_CURRENT_SOURCE_DIR}/${source}\": {\n")
+    file(APPEND ${FILEMAP} "\"dependencies\": \"${BUILD_DIR}/${source}.d\",\n")
```

Comment:
The .d style dependencies won't currently work without a way of telling CMake about them, and I am not sure if that is possible for a custom command.

We may need to simply make the command depend on all the public headers via explicit relationships, to get proper dependencies for when the public API is changed.